src/algebra/monoid_algebra/degree.lean

```diff
@@ -16,14 +16,15 @@ Currently, the only result is `degree_mul_le`: the degree of a product is at mos
 degrees. -/
 
 namespace add_monoid_algebra
+open_locale classical
 
-variables {R A : Type*} [semiring R] [linear_order A]
+variables {R A : Type*} [semiring R] [semilattice_sup A] [order_bot A]
 
 /--  Let `R` be a semiring, let `A` be a Type with a linear order, and let `f : R[A]` be an
 an element of `add_monoid_algebra R A`.  The degree of `f` takes values in `with_bot A` and
 it is the maximum of the support of `f` or `⊥`, depending on whether `f` is non-zero or not. -/
-def degree (f : add_monoid_algebra R A) : with_bot A :=
-f.support.max
+def degree (f : add_monoid_algebra R A) : A :=
+f.support.sup id
```

Comment:
I suspect you actually want `sup coe`, since otherwise you can't use this to define `polynomial.degree`

Reply:
Indeed, I was just experimenting with a grading type that had a bot.  I should have reverted to what I really meant.